.github/workflows/tests.yml

```diff
@@ -112,7 +112,11 @@ jobs:
         run: echo 'DISABLE_REMOTES_ARG=--disable-remotes=azure,gs' >> $env:GITHUB_ENV
 
       - name: Run tests
-        run: nox -s tests-${{ matrix.pyv }} -- $DISABLE_REMOTES_ARG
+        run: nox -s tests-${{ matrix.pyv }} -- -m "not e2e and not examples" $DISABLE_REMOTES_ARG
+        shell: bash
+
+      - name: Run E2E tests
+        run: nox -s tests-${{ matrix.pyv }} -- -m "e2e" $DISABLE_REMOTES_ARG
```

Comment:
[Q] Do we get some benefit from splitting this out?

Reply:
Will the tests be distributed amongst different workers?

Reply:
> [Q] Do we get some benefit from splitting this out?

I hope that this will make these issues obvious by splitting them.


> Will the tests be distributed amongst different workers?

Yes, it runs in the same way as the above test run.  PTAL at https://github.com/iterative/datachain/actions/runs/12880225335/job/35909694647#step:9:17 where this workflow is actually run (we use `pull_request_target` which is running workflow from the default branch).